code/frameworks/svelte-vite/src/plugins/ts2doc.ts

```diff
@@ -0,0 +1,209 @@
+import ts from 'typescript';
+import { svelte2tsx } from 'svelte2tsx';
+import { VERSION } from 'svelte/compiler';
+import { type SvelteDataItem } from 'sveltedoc-parser';
+
+function getInitializerValue(initializer?: ts.Node) {
+  if (initializer === undefined) {
+    return undefined;
+  }
+  if (ts.isNumericLiteral(initializer)) {
+    return Number(initializer.text);
+  }
+  if (ts.isStringLiteral(initializer)) {
+    return `"${initializer.text}"`;
+  }
+  if (initializer.kind === ts.SyntaxKind.TrueKeyword) {
+    return true;
+  }
+  if (initializer.kind === ts.SyntaxKind.FalseKeyword) {
+    return false;
+  }
+  return initializer.getText();
+}
+
+/*
+ * Make sveltedoc-parser compatible data from .svelte file with svelte2tsx and TypeScript.
+ */
+export function ts2doc(fileContent: string) {
```

Comment:
```suggestion
export function generateDocgen(fileContent: string) {
```

Filename also needs changing.